mdf_matio/__init__.py

```diff
@@ -158,11 +203,40 @@ def generate_search_index(directory: str, validate_records=True,
 
         # Loop over all produced records
         metadata = group.metadata if isinstance(group.metadata, list) else [group.metadata]
+        # Validate metadata and tweak into final MDF feedstock format
+        # Will fail if any entry fails validation - no invalid entries can be allowed
+        # TODO list:
+        #   - How should validation failures (which stop processing) be communicated?
+        #   - How should the feedstock be output? (Currently yield-ed, could be written to file)
```

Comment:
Maybe delegate that to another function. Keeping this function as a generator gives us flexibility to put whatever front-end we want on it.

Reply:
Refactored Validator continues to `yield`, as does `generate_search_index()`.